modules/writing_observer/writing_observer/awe_nlp.py

```diff
@@ -200,54 +202,129 @@ async def process_texts_parallel(texts, options=None):
     return annotated
 
 
-async def process_texts(writing_data, options=None, mode=RUN_MODES.MULTIPROCESSING):
+async def get_latest_cache(cache, text_hash):
+    """
+    Cache Helper Function: Returns latest cache for the text hash or initializes key-value pair for that hash if it does not already exist in the cache.
+    """
+    text_cache_data = await cache[text_hash]
+    if text_cache_data is None:
+        text_cache_data = {}
+    return text_cache_data
+
+
+async def check_features_available(cache, text_hash, set_options, writing):
+    """
+    Cache Helper Function : Check if some options are a subset of features_available
+    """
+    text_cache_data = await get_latest_cache(cache, text_hash)  # Get latest cache
+    found_options = set()
+    text_cache_data.setdefault('features_available', dict())
```

Comment:
You call `features_available` multiple times. You should define a variable and use that instead,

Reply:
Hey brad, could you clarify this with an example, I'm not exactly sure about this. @bradley-erickson

Reply:
```python
features_available = 'features_available'
# ...
text_cache_data.setdefault(features_available, dict())
```
Any string that is reused multiple times, such as `'features_available'` ought to be set to a variable so we do not accidentally introduce typos while working on this code later.